### PR TITLE
Update ESET Inspect connector (was: ESET Enterprise Inspector)

### PR DESCRIPTION
**Pull request by dmaasland**
Change(s):
   - Renames the connector from "ESET Enterprise Inspector" to "ESET Inspect"
   - Add support for the cloud version of ESET Inspect
   - No longer use queues to improve stability

   Reason for Change(s):
   - The product for which this connector can be used was renamed and a cloud version was introduced.

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hi @dmaasland, please add sample data of ESETInspect_CL in [sample data folder](https://github.com/dmaasland/Azure-Sentinel/tree/update-ei-connector/Sample%20Data/Custom), thanks.

**Comment by dmaasland** [comment]
@v-sabiraj done :).

File: DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        }
+      ]
+    },
+    {
+      "title": "Step 3 - Deploy the Azure Resource Manager (ARM) Template",
+      "description": "Use this method for automated deployment of the ESET Inspect connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-ESETInspect-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the **Workspace ID**, **Workspace Key**, **API Username**, **API Password** , enter the **Inspect base URL** and the **first ID** to start ingesting detections from.\n - The defailt starting ID is **0**. This means that all detections will be ingested. \n - Note: If using Azure Key Vault secrets for any of the values above, use the`@Microsoft.KeyVault(SecretUri={Security Identifier})`schema in place of the string values. Refer to [Key Vault references documentation](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) for further details. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (dmaasland) [reply DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json:137]: Sorry if this is a silly question, but would you happen to have an example of what kind of metadata needs to be added? Thanks.

File: DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }
+      ]
+    },
+    {
+      "title": "Step 3 - Deploy the Azure Resource Manager (ARM) Template",
+      "description": "Use this method for automated deployment of the ESET Inspect connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-ESETInspect-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the **Workspace ID**, **Workspace Key**, **API Username**, **API Password** , enter the **Inspect base URL** and the **first ID** to start ingesting detections from.\n - The defailt starting ID is **0**. This means that all detections will be ingested. \n - Note: If using Azure Key Vault secrets for any of the values above, use the`@Microsoft.KeyVault(SecretUri={Security Identifier})`schema in place of the string values. Refer to [Key Vault references documentation](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) for further details. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
```
Comment (v-sabiraj) [inline DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json:134]: @dmaasland, please fix the typo "labeled" to "labelled", thanks.

File: DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (v-sabiraj) [reply DataConnectors/ESET Inspect/ESETInspect_API_FunctionApp.json:137]: @dmaasland, please refer this [template ](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json), to add metadata, thanks.

File: Sample Data/Custom/ESETInspect_CL.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
     "TableName": "PrimaryResult",
     "TenantId": "7bdecb7b-0545-4aa9-bd6b-f7eb05fee857",
     "TimeGenerated": "2020-12-07T10:40:12.774Z",
-    "Type": "ESETEnterpriseInspector_CL",
+    "Type": "ESETInspect_CL",
     "_ResourceId": "",
     "computerId_d": "3",
```
Comment (v-sabiraj) [inline Sample Data/Custom/ESETInspect_CL.json:56]: Hey @dmaasland, as "computerId_d": "3", is added with "" it acts as string, please add the value as "computerId_d":  3, at all places, thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hey @dmaasland, can you please resolve the requested changes. Thanks.

**Comment by v-dvedak** [comment]
Hi @dmaasland, can you please resolve the requested changes? Thanks.

**Comment by dmaasland** [comment]
Alright, I think I have them all

**Review by v-sabiraj — APPROVED** [review]
looks good

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**